torchtitan/datasets/hf_datasets.py

```diff
@@ -79,32 +81,18 @@ def __init__(
                     f"Supported datasets are: {list(_supported_datasets.keys())}."
                 )
 
-        # special case to auto-load c4_mini (and any future datasets) from local dir
-        if dataset_name == "c4_mini":
-            dataset_path = f"torchtitan/datasets/{dataset_name}"
+        if not dataset_path:
+            dataset_path = _supported_datasets[dataset_name]
+        logger.info(f"Preparing {dataset_name} dataset from {dataset_path}")
 
-        # TODO: This is a temporary solution for small datasets.
-        #       For large datasets we need to use a more scalable approach,
-        #       and support shuffling and checkpointing.
-        if dataset_path:
-            logger.info(f"Loading {dataset_name} dataset locally from {dataset_path}")
-            ds = load_from_disk(dataset_path)
+        if dataset_name == "c4":
+            # c4 is huge, and requires both streaming and language selection
+            # (we default to en)
+            ds = load_dataset(dataset_path, name="en", split="train", streaming=True)
```

Comment:
can we add a `streaming` as an arg  the HuggingFaceDataset, and then when creating dataloader in c4, we just pass `streaming=true` to the HuggingFaceDataset constructor?

Reply:
We certainly can, although this would create if-c4 statements to other places as well and more args in function call in general. Since we only have two datasets right now, maybe let's keep it until the change is necessary.